src/networking.c

```diff
@@ -1583,10 +1583,11 @@ void unlinkClient(client *c) {
 
     /* Remove from the list of pending writes if needed. */
     if (c->flag.pending_write) {
-        serverAssert(&c->clients_pending_write_node.next != NULL || &c->clients_pending_write_node.prev != NULL);
         if (c->io_write_state == CLIENT_IDLE) {
+            serverAssert(listSearchKey(server.clients_pending_write, c) != NULL);
```

Comment:
do we have some other ways to assert it? I am worried that this search operation will be slow

Reply:
We can use debugServerAssert. It only enabled by the config `enable-debug-assert` and does nothing otherwise.

```suggestion
            debugServerAssert(listSearchKey(server.clients_pending_write, c) != NULL);
```

Reply:
Thanks, changed to debug assert.

Reply:
Found a better general way to assert node membership in the list using O(1) operation for all lists.